libs/entitlement/src/main/java/org/elasticsearch/entitlement/runtime/policy/PolicyManager.java

```diff
@@ -56,27 +84,120 @@ public PolicyManager(Policy defaultPolicy, Map<String, Policy> pluginPolicies, F
         this.pluginResolver = pluginResolver;
     }
 
-    public void checkFlagEntitlement(Class<?> callerClass, FlagEntitlementType type) {
+    private static List<Entitlement> lookupEntitlementsForModule(Policy policy, String moduleName) {
+        for (int i = 0; i < policy.scopes.size(); ++i) {
```

Comment:
Why are we scanning module names, this could be a map right?

Reply:
The map is lazily created, but it could be done at initialization time instead. I will move that in a follow up.